cpp/test/neighbors/ann_cagra.cuh

```diff
@@ -200,6 +202,69 @@ void GenerateRoundingErrorFreeDataset(
   GenerateRoundingErrorFreeDataset_kernel<T>
     <<<grid_size, block_size, 0, cuda_stream>>>(ptr, size, resolution);
 }
+
+template <class T>
+__global__ void Normalize8bitInt_kernel(T* const datatset_ptr,
+                                        const std::uint32_t size,
```

Comment:
Thank you for letting me know about the function, but I'm not sure it is possible to use it because, in my understanding, the function performs an element-wise operation for a given matrix. If my understanding of the function is incorrect, please let me know. (btw, I think the function name is very confusing because MatVec means a matrix-vector product in general but this is not...)

Reply:
Yes, it is preferred to use existing kernels. I see below that you are using `row_nolmalize` with float input. If the existing kernels do not work with `int8` types (and fixing them is non-trivial) then please open an Issue that describes the problem.

Reply:
I understand that it is better to use an existing kernel, but we can't use the `row_normalize` function for DataT=(u)int8 because the accumulator data type is the same as the input ((u)int8), and the norm calculation can overflow. I'm sorry I didn't explain why the `row_normalize` function is not used.
https://github.com/rapidsai/raft/blob/da3b9a9c442396a43a70efa725ca7f489605d632/cpp/include/raft/linalg/detail/normalize.cuh#L38-L63

Reply:
I fixed the normalization of a (u)int8 matrix to use `linalg::reduce` and `linalg::matrix_vector_op` instead of the CUDA kernel.